lib/api/model/model.dart

```diff
@@ -303,6 +315,11 @@ enum UserRole{
   final int? apiValue;
 
   int? toJson() => apiValue;
+
+  bool isAtLeast(UserRole threshold) {
+    // Roles with more privilege have lower [apiValue].
+    return (apiValue ?? 0) <= (threshold.apiValue ?? double.maxFinite);
```

Comment:
It doesn't make sense semantically for an "is at least" function to make up an answer like this does when one of the roles is unknown. When one role is unknown, the truth is that we don't know whether `this` is at least `threshold`.

(It could be done given the additional structure on unknown roles that Chris described at https://github.com/zulip/zulip-flutter/pull/886#discussion_r1720487156 and https://github.com/zulip/zulip-flutter/pull/886#discussion_r1731912511 . But as discussed there, it's not worth implementing that structure.)

Because this method can't determine the answer in that case, it should refuse to guess. The simplest way to do that is to have this method throw if either role is unknown, and make it the caller's job to check for that first and decide how to act on the uncertainty.